mods/ra/maps/soviet-05/main.lua

```diff
@@ -0,0 +1,212 @@
+IdleHunt = function(unit) if not unit.IsDead and not unit.Type == "harv" then Trigger.OnIdle(unit, unit.Hunt) end end
+
+CheckForBase = function()
+	baseBuildings = Map.ActorsInBox(Map.TopLeft, CFBPoint.CenterPosition, function(actor)
+		return actor.Type == "fact" or actor.Type == "powr"
+	end)
+
+	return #baseBuildings >= 2
+end
+
+CheckForCYard = function()
+	ConYard = Map.ActorsInBox(mcvGGLoadPoint.CenterPosition, ReinfEastPoint.CenterPosition, function(actor)
+		return actor.Type == "fact" and actor.Owner == GoodGuy
+	end)
+
+	return #ConYard >= 1
+end
+
+CheckForSPen = function()
+	SPens = Map.ActorsInBox(Map.TopLeft, Map.BottomRight, function(actor)
+		return actor.Type == "spen"
+	end)
+
+	return #SPens >=1
+end
+
+RunInitialActivities = function()
+	if Map.Difficulty == "Hard" then
+		Expand()
+		ExpansionCheck = true
+	else
+		ExpansionCheck = false
+	end
+
+	Trigger.AfterDelay(1, function(harvester)
+		Harvester.FindResources()
+		Helper.Destroy()
+	end)
+	
+	Reinforcements.Reinforce(player, sov1, sov1Path, 0, function(soldier)
+		soldier.AttackMove(SovietBasePoint.Location)
+	end)
+
+	Actor.Create("camera", true, { Owner = player, Location = GreeceBasePoint.Location })
+	Actor.Create("camera", true, { Owner = player, Location = SovietBasePoint.Location })
+
+	startmcv.Move(MCVStartMovePoint.Location)
+	Runner1.Move(RunnerRetreatPoint.Location)
+	Runner2.Move(RunnerRetreatPoint.Location)
+	Runner3.Move(RunnerRetreatPoint.Location)
+
+	ProduceInfantry()
+	Trigger.AfterDelay(DateTime.Minutes(2), ProduceShips)
+
+	Media.PlaySpeechNotification(player, "ReinforcementsArrived")
+
+	if Map.Difficulty == "Hard" or Map.Difficulty == "Medium" then
+		Trigger.AfterDelay(DateTime.Seconds(25), Gguy1)
+	end
+	Trigger.AfterDelay(DateTime.Minutes(2), Gguy1)
+	Trigger.AfterDelay(DateTime.Minutes(5), Gguy1)
+end
+
+Expand = function()
+	if not ExpansionCheck then
+		if not mcvtransport.IsDead then
+		mcvtransport.Move(mcvGGLoadPoint.Location)
+		end
+		if not mcvGG.IsDead then
+		mcvGG.Move(mcvGGLoadPoint.Location)
+		end
+		Reinforcements.Reinforce(GoodGuy, {"dd", "dd"}, {ReinfNorthPoint.Location, NearDockPoint.Location}, 0, function(ddsquad)
+			ddsquad.AttackMove(NearExpansionPoint.Location)
+		end)
+		ExpansionCheck = true
+		Trigger.ClearAll(mcvGG)
+		Trigger.ClearAll(mcvtransport)
+		Trigger.AfterDelay(DateTime.Seconds(3), function(load)
+			if not mcvtransport.IsDead then
```

Comment:
Remade, not sure if better or worse though. I like it more.
http://pastebin.com/7V4cMt57